packages/logging/src/logging-and-telemetry.ts

```diff
@@ -52,10 +52,23 @@ import type {
   MongoshLoggingAndTelemetryArguments,
   MongoshTrackingProperties,
 } from './types';
+import { machineIdSync } from 'node-machine-id';
 
 export function setupLoggingAndTelemetry(
   props: MongoshLoggingAndTelemetryArguments
 ): MongoshLoggingAndTelemetry {
+  if (!props.deviceId) {
```

Comment:
Seemed more logical for me for the device Id determination to exist here rather than in the class itself as then its state is immutable